connector/exceptionsconnector/connector.go

```diff
@@ -0,0 +1,348 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exceptionsconnector
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/tilinna/clock"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/exceptionsconnector/internal/cache"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/traceutil"
+)
+
+const (
+	serviceNameKey     = conventions.AttributeServiceName
+	spanKindKey        = "span.kind"   // OpenTelemetry non-standard constant.
+	statusCodeKey      = "status.code" // OpenTelemetry non-standard constant.
+	metricKeySeparator = string(byte(0))
+
+	defaultDimensionsCacheSize = 1000
+)
+
+type metricKey string
+
+type connectorImp struct {
+	lock   sync.Mutex
+	logger *zap.Logger
+	config Config
+
+	metricsConsumer consumer.Metrics
+
+	// Additional dimensions to add to metrics.
+	dimensions []dimension
+
+	// The starting time of the data points.
+	startTimestamp pcommon.Timestamp
+
+	keyBuf *bytes.Buffer
+
+	exceptions map[metricKey]int
+	// An LRU cache of dimension key-value maps keyed by a unique identifier formed by a concatenation of its values:
+	// e.g. { "foo/barOK": { "serviceName": "foo", "status_code": "OK" }}
+	metricKeyToDimensions *cache.Cache[metricKey, pcommon.Map]
+
+	ticker  *clock.Ticker
```

Comment:
Do we really need to use a ticker? Can we leave aggregation as well to a processor in the next pipeline? What if this connector just pushed data over to metrics as traces came in?